### ISSUE TITLE
Login with native facebook app not working (Firebase)

### ISSUE THREAD

**Issue by PiroX4256** [darwin-morocho/flutter-facebook-auth#211]
**Describe the bug**
Hi, I am using the latest version (4.0.1) of this plugin in order to authenticate through firebase using facebook accounts.
I have an issue on iOS. When I tap on facebook login, it tries to use the native Facebook app to login. It switches to it and then back to my application, but the sign in procedure is not called anymore.

I can see this response in my console:

`iOS handleLink: fb1288553841625957://authorize#denied_scopes=&granted_scopes=email%2Copenid%2Cpublic_profile&graph_domain=facebook&nonce=AWXNbKFx-REJYToWl_1hCajg-GGHPvwBjuSKEWtNbhaAPH1dkCYh_tzTxv-i0ssjicSSPcA-W2221uKlofXzkmGjLbWUpIdlq80&signed_request=FLpzkuCAHoyniD9SkZJsiqV0QJ65A9aiZDjNFmJowfc.eyJ1c2VyX2lkIjoiNTAxMDQ1MzI3NTY0NjEwMCIsImNvZGUiOiJBUUNHVnZGMEdlQlp1T0ZPRm5veGlkZ0hTYVRqOS1BaEFOUFBOZ3pIZzBuTmZNb1dTOVJIM2N5TUVLbGk4MUVoM1FVZmZQUTBNSDJJdnZRYnQ0Yy1sOXNrUkNPSktKSl9NV05jYnQ2aDF0b3RUekdXMTlHZHJlR1ZBT3pnRUg1Tkt1WENMMVZGZ21CV25uQUt2MFczU3ZmVzczQlpYRldYWVBsMExKdElzVlVQUmxHRnBUQ1dJWk50d1p2Z2xNM3dXdDZmU3E4WWdqYUlOMzNzTEw2MDZKTElPOFRkZ2o1Qy15QWNWcGM0cUFybFVRcExJckh3bXhQZkx2b1VCakgxelhSV1E4V3h2S2JqMG02bzNoUzVqV20wN2VxZXlpU0xfRE8tVTZNQVlpQXpHYTRnanN6c3NUdEVOcEcyaWhNQi1INGUyUnVqTUlPNkxLNmdZVnZTamp2OEJCZTE3ZmJDQ2VuUURCZ015OE9YcGRYeGhJUC1TWWNsRVdRaDl0akhUblM2SnRzTlRSYUo1TWREN1VmMmM3b3kiLCJhbGdvcml0aG0iOiJITUFDLVNIQTI1NiIsImlzc3VlZF9hdCI6MTY0MjE2MDIzNn0&state=%7B%22challenge%22%3A%22%253Dt6G%252F0057pqYCpyt%253DonXUy1n3K8%253D%22%2C%220_auth_logger_id%22%3A%22076D215F-0B10-430B-8DA3-632B8139841E%22%2C%22com.facebook.sdk_client_state%22%3Atrue%2C%223_method%22%3A%22sfvc_auth%22%7D"`

**Environment**
My Info.plist, obv I hide the secrets.

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleDevelopmentRegion</key>
	<string>$(DEVELOPMENT_LANGUAGE)</string>
	<key>CFBundleDisplayName</key>
	<string>$(FLAVOR_APP_NAME)</string>
	<key>CFBundleExecutable</key>
	<string>$(EXECUTABLE_NAME)</string>
	<key>CFBundleIdentifier</key>
	<string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundleLocalizations</key>
	<array>
		<string>en</string>
		<string>es</string>
	</array>
	<key>CFBundleName</key>
	<string>Hasks</string>
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>CFBundleShortVersionString</key>
	<string>$(FLUTTER_BUILD_NAME)</string>
	<key>CFBundleSignature</key>
	<string>????</string>
	<key>CFBundleURLTypes</key>
    <array>
        <dict>
            <key>CFBundleURLSchemes</key>
            <array>
                <string>APP-ID</string>
                <string>hasks-firebase</string>
            </array>
        </dict>

        <dict>
            <key>CFBundleTypeRole</key>
            <string>Editor</string>
            <key>CFBundleURLSchemes</key>
            <array>
                <string>URL-SCHEME</string>
            </array>
        </dict>
    </array>
    <key>FacebookAppID</key>
    <string>APP-ID</string>
    <key>FacebookClientToken</key>
    <string>CLIENT-TOKEN</string>
    <key>FacebookDisplayName</key>
    <string>Hasks</string>

    <key>LSApplicationQueriesSchemes</key>
    <array>
      <string>fbapi</string>
      <string>fbapi20130214</string>
      <string>fbapi20130410</string>
      <string>fbapi20130702</string>
      <string>fbapi20131010</string>
      <string>fbapi20131219</string>
      <string>fbapi20140410</string>
      <string>fbapi20140116</string>
      <string>fbapi20150313</string>
      <string>fbapi20150629</string>
      <string>fbapi20160328</string>
      <string>fbauth</string>
      <string>fb-messenger-share-api</string>
      <string>fbauth2</string>
      <string>fbshareextension</string>
    </array>

	<key>CFBundleVersion</key>
	<string>$(FLUTTER_BUILD_NUMBER)</string>
	<key>FacebookAppID</key>
	<string>1288553841625957</string>
	<key>FacebookClientToken</key>
    <string>a2e7b55e6c560bf8dcf8dc25fa921265</string>
	<key>FacebookDisplayName</key>
	<string>Hasks</string>
	<key>FirebaseAppDelegateProxyEnabled</key>
	<false/>
	<key>LSRequiresIPhoneOS</key>
	<true/>
	<key>NSCameraUsageDescription</key>
	<string>Pick image for groups and profiles</string>
	<key>NSMicrophoneUsageDescription</key>
	<string>Pick image for groups and profiles</string>
	<key>NSPhotoLibraryUsageDescription</key>
	<string>Pick image for groups and profiles</string>
	<key>UIBackgroundModes</key>
	<array>
		<string>fetch</string>
		<string>remote-notification</string>
	</array>
	<key>UILaunchStoryboardName</key>
	<string>LaunchScreen</string>
	<key>UIMainStoryboardFile</key>
	<string>Main</string>
	<key>UIStatusBarHidden</key>
	<true/>
	<key>UISupportedInterfaceOrientations</key>
	<array>
		<string>UIInterfaceOrientationPortrait</string>
		<string>UIInterfaceOrientationLandscapeLeft</string>
		<string>UIInterfaceOrientationLandscapeRight</string>
	</array>
	<key>UISupportedInterfaceOrientations~ipad</key>
	<array>
		<string>UIInterfaceOrientationPortrait</string>
		<string>UIInterfaceOrientationPortraitUpsideDown</string>
		<string>UIInterfaceOrientationLandscapeLeft</string>
		<string>UIInterfaceOrientationLandscapeRight</string>
	</array>
	<key>UIViewControllerBasedStatusBarAppearance</key>
	<false/>
	<key>FacebookAutoLogAppEventsEnabled</key>
    <true/>
</dict>
</plist>
```

**Comment by PiroX4256** [comment]
I forgot my AppDelegate configuration, which is the one provided by firebase and works with all other IdPs.

@UIApplicationMain
class AppDelegate: FlutterAppDelegate {
    override func application(
        _ application: UIApplication,
        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil
    ) -> Bool
    {
        FirebaseApp.configure()
        GeneratedPluginRegistrant.register(with: self)
        return super.application(application, didFinishLaunchingWithOptions: launchOptions) // YES;
    }
}

**Comment by PiroX4256** [comment]
[FIXED]
Solved by reading issue #6 and modifying my AppDelegate.swift in this way

```
@UIApplicationMain
class AppDelegate: FlutterAppDelegate {
    override func application(
        _ application: UIApplication,
        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil
    ) -> Bool
    {
        FirebaseApp.configure()
        GeneratedPluginRegistrant.register(with: self)
        return super.application(application, didFinishLaunchingWithOptions: launchOptions) // YES;
    }

    override func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
            if url.scheme != nil {
                let facebookAppId: String? = Bundle.main.object(forInfoDictionaryKey: "FacebookAppID") as? String
                if facebookAppId != nil && url.scheme!.hasPrefix("fb\(facebookAppId!)") && url.host ==  "authorize" {
                     print("is login by facebook")
                     return ApplicationDelegate.shared.application(app, open: url, options: options)
                }
            }

            return false

        }
}
```

I suggest to update documentation for Firebase case, by specifying this behavior.

**Comment by ishaquehassan** [comment]
Adding to @PiroX4256 answer, also make sure to add
`import FBSDKCoreKit`
at the top of your `AppDelegate.swift` 

Hope it helps
